src/sql/azdata.proposed.d.ts

```diff
@@ -644,13 +654,27 @@ declare module 'azdata' {
 
 		export type DialogWidth = 'narrow' | 'medium' | 'wide' | number;
 
+		export type DialogStyle = 'normal' | 'flyout' | 'callout' | 'calloutCompact';
+
+		export type DialogPosition = 'left' | 'right' | 'above' | 'below' | undefined;
+
+		export interface IDialogXYOffset {
+			xOffset: number;
+			yOffset: number;
+		}
+
 		/**
 		 * Create a dialog with the given title
 		 * @param title The title of the dialog, displayed at the top
 		 * @param dialogName the name of the dialog
 		 * @param width width of the dialog, default is 'wide'
+		 * @param dialogStyle Defines the dialog style, default is 'Flyout'
+		 * @param dialogPosition Defines the dialog position, default is null
+		 * @param suppressHeader Prevents render of Dialog header
```

Comment:
So with these `suppress` params : 

1. Not a fan of `suppress` for the name - just having it be `renderHeader` and be set to false if the user doesn't want to render it seems fine
2. I assume this is meant for use by the callout dialog since typically that wouldn't have a header/footer. In which case why wouldn't we just do that automatically if the dialog is a callout? I could see us in the future wanting to keep the header/footer for some callout dialogs but do you actually need that feature now? If not then I'd say let's keep this simple and just default to not showing for callout.